src/step-function.ts

```diff
@@ -466,6 +466,428 @@ export namespace $SFN {
+        const dataOut = evalExprToJsonPathOrLiteral(data.expr);
+
+        if (
+          ASLGraph.isLiteralValue(dataOut) &&
+          typeof dataOut.value !== "string"
+        ) {
+          throw new SynthError(
+            ErrorCodes.Invalid_Input,
+            "Expected base64Encode data argument to be a string or reference."
+          );
+        }
+        const temp = context.newHeapVariable();
+        return {
+          Type: "Pass",
+          Parameters: {
+            "out.$": `States.Base64Encode(${
+              ASLGraph.isJsonPath(dataOut)
+                ? dataOut.jsonPath
+                : `'${dataOut.value}'`
+            })`,
+          },
+          ResultPath: temp,
+          Next: ASLGraph.DeferNext,
+          output: {
+            jsonPath: `${temp}.out`,
+          },
+        };
+      });
+    },
+  });
+
+  /**
+   * Use the base64Decode intrinsic function to decode data based on MIME Base64 decoding scheme.
+   * You can use this function to pass data to other AWS services without using a Lambda function.
+   *
+   * This function takes a data string of up to 10,000 characters to encode as its only argument.
+   *
+   * @param base64 - Base64 string to decode.
+   */
+  export const base64Decode = makeStepFunctionIntegration<
+    "SFN.Base64Decode",
+    (base64: string) => string
+  >("SFN.Base64Decode", {
+    asl(call, context) {
+      const [data] = call.args;
+
+      if (!data) {
+        throw new SynthError(
+          ErrorCodes.Invalid_Input,
+          "Expected Base64 data argument to be provided."
+        );
+      }
+
+      return context.evalContext(call, ({ evalExprToJsonPathOrLiteral }) => {
+        const dataOut = evalExprToJsonPathOrLiteral(data.expr);
+
+        if (
+          ASLGraph.isLiteralValue(dataOut) &&
+          typeof dataOut.value !== "string"
+        ) {
+          throw new SynthError(
+            ErrorCodes.Invalid_Input,
+            "Expected base64Decode data argument to be a string or reference."
+          );
+        }
+        const temp = context.newHeapVariable();
+        return {
+          Type: "Pass",
+          Parameters: {
+            "out.$": `States.Base64Decode(${
+              ASLGraph.isJsonPath(dataOut)
+                ? dataOut.jsonPath
+                : `'${dataOut.value}'`
+            })`,
+          },
+          ResultPath: temp,
+          Next: ASLGraph.DeferNext,
+          output: {
+            jsonPath: `${temp}.out`,
+          },
+        };
+      });
+    },
+  });
+
+  export type HashAlgorithm =
+    | "MD5"
+    | "SHA-1"
+    | "SHA-256"
+    | "SHA-384"
+    | "SHA-512";
+
+  /**
+   * Use the `hash` intrinsic function to calculate the hash value of a given input.
+   * You can use this function to pass data to other AWS services without using a Lambda function.
+   *
+   * @param data - data to hash.
+   * @param algorithm - algorithm to use.
+   */
+  export const hash = makeStepFunctionIntegration<
```

Comment:
I've been playing with the docs. The make integration utility creates problematic types that don't render well. 

A half solution is to add an explicit type signature on the variable here for the function. This is better but it still has the problem of being classified as a property instead of a method. 

The best solution is to have an interface with each of these functions defined as methods with documentation. Then have this implementation be internal and hidden.

Reply:
Have an example?

Reply:
```ts
interface Intrinsics {
  random(..): number
}

const Intrinsics: Intrinsics = {
  random: makeIntegrstion(..)
}
```

Reply:
Ts docs work off of type signatures. We don't need the integration signature in there and we want to separate functions from properties.

Reply:
this works, by the way, at least in the IDE

![image](https://user-images.githubusercontent.com/289213/187959275-23503826-0be5-4ad6-be97-bfe3a91d3489.png)

![image](https://user-images.githubusercontent.com/289213/187959341-82167170-005f-4d2f-89bc-9588647c8c6c.png)

Reply:
Alright, updated them all to the interface + const pattern.

Looks nice.

Reply:
![image](https://user-images.githubusercontent.com/289213/187964282-494a6821-f594-4938-bb84-96754d0ff57d.png)